File: py110/small_problems_110_119/medium2/6.py
# https://launchschool.com/exercises/c5b25237

def sum_square_difference(number):
    sum_before_square = 0
    square_by_number = 0

    for num in range(1, number + 1):
        sum_before_square += num
        square_by_number += (num **2)

    return (sum_before_square**2) - square_by_number


print(sum_square_difference(3) == 22)          # True
# 22 --> (1 + 2 + 3)**2 - (1**2 + 2**2 + 3**2)

print(sum_square_difference(10) == 2640)       # True
print(sum_square_difference(1) == 0)           # True
print(sum_square_difference(100) == 25164150)  # True

File: ls215/smallProblems/med2/6.js
// https://launchschool.com/exercises/ac05fdea

'use strict';

const sumSquareDifference = (num) => {
  let sum = 0;
  let squaredSum = 0

  for (let i = 1; i <= num; i += 1) {
    sum += i;
    squaredSum += (i ** 2);
  }

  return (sum ** 2) - squaredSum;
};



console.log(sumSquareDifference(3));      // 22 --> (1 + 2 + 3)**2 - (1**2 + 2**2 + 3**2)
console.log(sumSquareDifference(10));     // 2640
console.log(sumSquareDifference(1));      // 0
console.log(sumSquareDifference(100));    // 25164150